### DISCUSSION TITLE
theme

### DISCUSSION THREAD

**Discussion by fadri14** [cmus/cmus#1266] [General]
Hello, can you please tell me how to change cmus theme via script? I have tried several approaches but nothing works perfectly. For example, I tried to retrieve the values ​​of a particular theme and change them directly in the autosave file, but it didn't work completely. Thanks to anyone who takes the time to help me.

**Comment by gavtroy** [comment]
You can do `:colorscheme amazon`, is that what you're asking?

If you mean creating your own theme, you can copy an existing .theme file to your cmus config directory and start experimenting with a text editor. Just make sure to keep the ".theme" file extension.

**Reply by fadri14** [reply]
I'd like to be able to alternate between two predefined themes using a script. So I can't use :colorscheme amazon because it has to be interpreted by cmus directly. Would you have another solution, please?

**Reply by gavtroy** [reply]
If cmus is running you can send the command immediately with `cmus-remote -C "colorscheme amazon"`. You can also set the individual color variables this way. Unfortunately there's no way to retrieve the name of the currently applied theme since cmus doesn't store it. The closest you could get is querying the individual variables like `cmus-remote -C "set color_win_bg?"`.

By the way, if cmus is closed then adding "`colorscheme amazon`" to the end of `autosave` should work. And if you want cmus to reset to a certain theme every time it starts, you can create a new file `rc` beside `autosave` and put the line in there.

**Reply by fadri14** [reply]
Ok thank you very much I have everything I need to do what I wanted. Really thank you for your help :)
